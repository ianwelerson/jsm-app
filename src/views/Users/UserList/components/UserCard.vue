<script setup lang="ts"></script>

<template>UserCard</template>

<style lang="scss" scoped></style>
