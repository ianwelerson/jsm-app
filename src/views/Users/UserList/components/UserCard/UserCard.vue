<script setup lang="ts">
export interface UserCardProps {
  id: string
  picture: string
  name: string
  street: string
  city: string
  state: string
  postcode: number
}

const props = defineProps<UserCardProps>()
</script>

<template>
  <RouterLink
    :to="{ name: 'userDetails', params: { userId: props.id } }"
    class="user-card"
  >
    <div class="user-card__wrapper">
      <div class="user-card__profile">
        <img
          :src="props.picture"
          :alt="`Foto de ${props.name}`"
          class="user-card__picture"
        />
      </div>
      <p class="user-card__name">{{ props.name }}</p>
      <p class="user-card__street">{{ props.street }}</p>
      <p class="user-card__city">{{ props.city }}</p>
      <p class="user-card__state-zip">
        {{ props.state }} - CEP: {{ props.postcode }}
      </p>
    </div>
  </RouterLink>
</template>

<style lang="scss" scoped>
.user-card {
  border: $border-size-base solid $gray-400;
  border-radius: $radius-base;
  display: flex;
  padding: $spacing-9 $spacing-3;
  text-decoration: none;
  transition: box-shadow 0.4s ease-in;

  &:hover {
    box-shadow: 2px 6px 12px 6px rgba(229, 229, 229, 0.57);
    transition: box-shadow 0.4s ease-out;
  }

  &__wrapper {
    align-items: center;
    display: flex;
    flex-direction: column;
    flex: 1;
  }

  &__profile {
    border: $border-size-base solid $gray-300;
    border-radius: $radius-full;
    height: 80px;
    overflow: hidden;
    width: 80px;
  }

  &__picture {
    height: 100%;
    object-fit: contain;
    width: 100%;
  }

  &__name,
  &__street,
  &__city,
  &__state-zip {
    color: $text-color;
    font-weight: $font-regular;
    text-transform: capitalize;
    text-align: center;
  }

  &__name {
    font-size: $text-lg;
    font-weight: $font-bold;
    line-height: $leading-3;
    margin-top: $spacing-2;
  }

  &__street {
    font-size: $text-sm;
    line-height: $leading-2;
    margin-top: $spacing-3;
  }

  &__city,
  &__state-zip {
    font-size: $text-xs;
    line-height: $leading-1;
  }

  &__city {
    margin-top: $spacing-2;
  }

  &__state-zip {
    margin-top: 0;
  }
}
</style>
