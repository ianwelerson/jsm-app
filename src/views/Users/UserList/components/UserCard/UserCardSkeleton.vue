<script setup lang="ts"></script>

<template>
  <div class="user-card-skeleton">
    <div class="user-card-skeleton__wrapper">
      <div class="user-card-skeleton__profile">
        <div class="user-card-skeleton__picture"></div>
      </div>
      <div class="user-card-skeleton__name"></div>
      <div class="user-card-skeleton__street"></div>
      <div class="user-card-skeleton__city"></div>
      <div class="user-card-skeleton__state-zip"></div>
    </div>
  </div>
</template>

<style lang="scss" scoped>
.user-card-skeleton {
  border: $border-size-base solid $gray-400;
  border-radius: $radius-base;
  display: flex;
  padding: $spacing-9 $spacing-3;

  &__wrapper {
    align-items: center;
    display: flex;
    flex-direction: column;
    flex: 1;
  }

  &__profile {
    border-radius: $radius-full;
    height: 80px;
    overflow: hidden;
    width: 80px;
  }

  &__picture {
    @include skeleton;
    height: 100%;
    width: 100%;
  }

  &__name,
  &__street,
  &__city,
  &__state-zip {
    height: 20px;
    margin-top: $spacing-2;
    width: 80%;
    @include skeleton;
  }

  &__street {
    margin-top: $spacing-3;
  }

  &__state-zip {
    margin-top: 2px;
  }
}
</style>
