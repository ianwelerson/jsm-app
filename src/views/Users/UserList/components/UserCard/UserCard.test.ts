import { render } from '@testing-library/vue'
import UserCard from '@/views/Users/UserList/components/UserCard/UserCard.vue'

const userDataMock = {
  id: 'alejandra.rodrigues',
  picture: 'https://randomuser.me/api/portraits/med/women/18.jpg',
  name: 'alejandra rodrigues',
  street: '3833 rua santa catarina',
  city: 'umuarama',
  state: 'santa catarina',
  postcode: 43646,
}

describe('UserCard', () => {
  it('should render all user information', async () => {
    const { getByText, getByRole } = render(UserCard, {
      props: {
        ...userDataMock,
      },
    })

    expect((getByRole('img') as HTMLImageElement).src).toBe(
      userDataMock.picture
    )
    getByText(userDataMock.name)
    getByText(userDataMock.street)
    getByText(userDataMock.city)
    getByText(`${userDataMock.state} - CEP: ${userDataMock.postcode}`)
  })

  it('should match the snapshot', async () => {
    const { html } = render(UserCard)

    expect(html()).toMatchSnapshot()
  })
})
