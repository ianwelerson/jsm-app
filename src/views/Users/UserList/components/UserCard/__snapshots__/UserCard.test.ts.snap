// Vitest Snapshot v1

exports[`UserCard > should match the snapshot 1`] = `
"<routerlink to=\\"[object Object]\\" class=\\"user-card\\" data-v-671e15bc=\\"\\">
  <div class=\\"user-card__wrapper\\" data-v-671e15bc=\\"\\">
    <div class=\\"user-card__profile\\" data-v-671e15bc=\\"\\"><img alt=\\"Foto de undefined\\" class=\\"user-card__picture\\" data-v-671e15bc=\\"\\"></div>
    <p class=\\"user-card__name\\" data-v-671e15bc=\\"\\"></p>
    <p class=\\"user-card__street\\" data-v-671e15bc=\\"\\"></p>
    <p class=\\"user-card__city\\" data-v-671e15bc=\\"\\"></p>
    <p class=\\"user-card__state-zip\\" data-v-671e15bc=\\"\\"> - CEP: </p>
  </div>
</routerlink>"
`;
